nautobot/dcim/forms.py

```diff
@@ -886,48 +898,158 @@ class DeviceTypeFilterForm(NautobotFilterForm):
+
+    class Meta:
+        model = ModuleType
+        fields = [
+            "manufacturer",
+            "model",
+            "part_number",
+            "tags",
+        ]
+
+
+# TODO
+# class ModuleTypeImportForm(BootstrapMixin, forms.ModelForm):
+#     """
+#     Form for JSON/YAML import of ModuleType objects.
+
+#     TODO: at some point we'll want to add general-purpose YAML serialization/deserialization,
+#     similar to what we've done for CSV in 2.0, but for the moment we're leaving this as-is so that we can remain
+#     at least nominally compatible with the netbox-community/devicetype-library repo.
+#     """
+
+#     manufacturer = forms.ModelChoiceField(queryset=Manufacturer.objects.all(), to_field_name="name")
+#     device_family = forms.ModelChoiceField(queryset=DeviceFamily.objects.all(), to_field_name="name", required=False)
+
+#     class Meta:
+#         model = ModuleType
+#         fields = [
+#             "manufacturer",
+#             "device_family",
+#             "model",
+#             "part_number",
+#             "u_height",
+#             "is_full_depth",
+#             "subdevice_role",
+#             "comments",
+#         ]
+
+
+class ModuleTypeBulkEditForm(TagsBulkEditFormMixin, NautobotBulkEditForm):
+    pk = forms.ModelMultipleChoiceField(queryset=ModuleType.objects.all(), widget=forms.MultipleHiddenInput())
+    manufacturer = DynamicModelChoiceField(queryset=Manufacturer.objects.all(), required=False)
+
+
+class ModuleTypeFilterForm(NautobotFilterForm):
+    model = ModuleType
+    q = forms.CharField(required=False, label="Search")
+    manufacturer = DynamicModelMultipleChoiceField(
+        queryset=Manufacturer.objects.all(), to_field_name="name", required=False
+    )
+    has_console_ports = forms.NullBooleanField(
+        required=False,
+        label="Has console ports",
+        widget=StaticSelect2(choices=BOOLEAN_WITH_BLANK_CHOICES),
+    )
+    has_console_server_ports = forms.NullBooleanField(
+        required=False,
+        label="Has console server ports",
+        widget=StaticSelect2(choices=BOOLEAN_WITH_BLANK_CHOICES),
+    )
+    has_power_ports = forms.NullBooleanField(
+        required=False,
+        label="Has power ports",
+        widget=StaticSelect2(choices=BOOLEAN_WITH_BLANK_CHOICES),
+    )
+    has_power_outlets = forms.NullBooleanField(
+        required=False,
+        label="Has power outlets",
+        widget=StaticSelect2(choices=BOOLEAN_WITH_BLANK_CHOICES),
+    )
+    has_interfaces = forms.NullBooleanField(
+        required=False,
+        label="Has interfaces",
+        widget=StaticSelect2(choices=BOOLEAN_WITH_BLANK_CHOICES),
+    )
+    tags = TagFilterField(model)
+
+
 #
 # Device component templates
 #
 
 
+class ComponentTemplateForm(NautobotModelForm):
+    # TODO: placeholder values shouldn't be form controls, instead use:
+    #       <p class="form-control-static">{{ obj|hyperlinked_object_target_new_tab }}</p>
```

Comment:
I need to circle back to this because it proved to be complicated to try to do it the "right" way (in the form field) and I might just have to cheat and do it in the html template for now.